File: src/models/entities/Phallic.java
package models.entities;

/**
 * Objects of Subclasses of this class are able to penetrate Fuckable BodyParts
 */
public interface Phallic {
    /**
     * size in cm
     */
    int size = 0;
    /**
     * width in mm
     */
    int width = 0;

    public int getSize();

    public int getWidth();
}

File: Assets/Scripts/entities/Phallic.cs
namespace entities {

	/**
	 * Objects of Subclasses of this class are able to penetrate Fuckable BodyParts
	 */
	public interface Phallic {
        /**
	     * name for identification
         */
        string name {
            get;
        }
        /**
	     * size in cm
	     */
        int size {
			get;
			set;
        }
        /**
	     * width in mm
	     */
        int width {
            get;
            set;
        }

        bool startPenetrating(Fuckable orifice);
        bool stopPenetrating(Fuckable orifice);

        bool Equals(object obj);
        int GetHashCode();
    }
}
